src/Microsoft.Health.Fhir.SqlServer/Features/Schema/Sql/Sprocs/UpsertResource_7.sql

```diff
@@ -119,50 +125,39 @@ FROM   dbo.Resource WITH (UPDLOCK, HOLDLOCK)
 WHERE  ResourceTypeId = @resourceTypeId
        AND ResourceId = @resourceId
        AND IsHistory = 0;
-IF (@etag IS NOT NULL
-    AND @etag <> @previousVersion)
-    BEGIN
-        THROW 50412, 'Precondition failed', 1;
-    END
+
 DECLARE @version AS INT; -- the version of the resource being written
 IF (@previousResourceSurrogateId IS NULL)
     BEGIN
-        
-		-- There is no previous version of this resource
-		IF (@isDeleted = 1)
-            BEGIN
-                -- Don't bother marking the resource as deleted since it already does not exist.
-				COMMIT TRANSACTION;
-                RETURN;
-            END
-        IF (@etag IS NOT NULL)
-            BEGIN
-                -- You can't update a resource with a specified version if the resource does not exist
-				THROW 50404, 'Resource with specified version not found', 1;
-            END
-        IF (@allowCreate = 0)
-            BEGIN
-                THROW 50405, 'Resource does not exist and create is not allowed', 1;
-            END
+        -- There is no previous version
         SET @version = 1;
```

Comment:
>SET @version = 1; [](http://example.com/codeflow?start=8&length=17)

@import sets initial version to 0. It is a discrepancy.

Reply:
Discussed this with Brendan, I will talk with the Import team to figure out if it's possible to start the version from 1 instead of 0

Reply:
Why is 1 required?

Reply:
Discussed this with Tong, He is going to help make this consistent for import functionality. Small note - SQL always started with version 1 from the beginning. Cosmos also starts with version 1